src/Explorer/Panes/SettingsPane/index.tsx

```diff
@@ -0,0 +1,408 @@
+import { PrimaryButton } from "office-ui-fabric-react";
+import React, { FunctionComponent, KeyboardEvent, useEffect, useLayoutEffect, useRef, useState } from "react";
+import * as Constants from "../../../Common/Constants";
+import { configContext } from "../../../ConfigContext";
+import { LocalStorageUtility, StorageKey } from "../../../Shared/StorageUtility";
+import * as StringUtility from "../../../Shared/StringUtility";
+import { userContext } from "../../../UserContext";
+import * as NotificationConsoleUtils from "../../../Utils/NotificationConsoleUtils";
+import { Tooltip } from "./Tooltip";
+
+export interface SettingsPaneProps {
+  closePanel: () => void;
+  openNotificationConsole: () => void;
+}
+
+export const SettingsPane: FunctionComponent<SettingsPaneProps> = ({
+  closePanel,
+  openNotificationConsole,
+}: SettingsPaneProps) => {
+  const [formErrors, setFormErrors] = useState("");
+  const [formErrorsDetails, setFormErrorsDetails] = useState();
+  const [isExecuting, setIsExecuting] = useState(false);
+  const [pageOption, setPageOption] = useState("");
+  const [customItemPerPage, setCustomItemPerPage] = useState(0);
+  const [crossPartitionQueryEnabled, setCrossPartitionQueryEnabled] = useState(
+    LocalStorageUtility.hasItem(StorageKey.IsCrossPartitionQueryEnabled)
+      ? LocalStorageUtility.getEntryString(StorageKey.IsCrossPartitionQueryEnabled) === "true"
+      : true
+  );
+  const [graphAutoVizDisabled, setGraphAutoVizDisabled] = useState(
+    LocalStorageUtility.hasItem(StorageKey.IsGraphAutoVizDisabled)
+      ? LocalStorageUtility.getEntryString(StorageKey.IsGraphAutoVizDisabled)
+      : "false"
+  );
+  const [maxDegreeOfParallelism, setMaxDegreeOfParallelism] = useState(
+    LocalStorageUtility.hasItem(StorageKey.MaxDegreeOfParellism)
+      ? LocalStorageUtility.getEntryNumber(StorageKey.MaxDegreeOfParellism)
+      : Constants.Queries.DefaultMaxDegreeOfParallelism
+  );
+  const explorerVersion = configContext.gitSha;
+  const shouldShowQueryPageOptions = userContext.apiType === "SQL";
+  const shouldShowGraphAutoVizOption = userContext.apiType === "Gremlin";
+  const shouldShowCrossPartitionOption = userContext.apiType !== "Gremlin";
+  const shouldShowParallelismOption = userContext.apiType !== "Gremlin";
+
+  const pageOptionsRef = useRef<HTMLLabelElement>();
+  const displayQueryRef = useRef<HTMLLabelElement>();
+  const maxDegreeRef = useRef<HTMLInputElement>();
+
+  useEffect(() => {
+    _loadSettings();
+  }, []);
+
+  useLayoutEffect(() => {
+    if (userContext.apiType === "Gremlin") {
+      displayQueryRef?.current?.focus();
+    } else if (userContext.apiType === "Tables") {
+      maxDegreeRef?.current?.focus();
+    }
+    pageOptionsRef?.current?.focus();
+  }, []);
```

Comment:
Remove this entirely. It think we don't need it anymore. The default focus behavior should be fine

Reply:
done